homeassistant/helpers/intent.py

```diff
@@ -294,20 +337,35 @@ def async_match_states(
             if _is_device_class(state, entity, device_classes)
         ]
 
+    filter_areas: list[area_registry.AreaEntry] = []
+
+    if (floor is None) and (floor_name is not None):
+        # Look up floor by name
+        floor = _find_floor(floor_name, floors)
+        if floor is None:
+            _LOGGER.warning("Floor not found: %s", floor_name)
```

Comment:
Shouldn't we raise like we did before (by abusing `assert`)

Reply:
No, I removed the asserts since the guarding is happening in the intent handlers above. It seemed odd that `async_match_states` would care about misnamed areas/floors rather than just return an empty list indicating there was no match.